test/api/projects_api_test.rb

```diff
@@ -11,10 +11,105 @@ def app
   end
 
   def test_can_get_projects
-    user = FactoryBot.create(:user, :student, enrol_in: 0)
-    get with_auth_token('/api/projects', user)
+    # Create a new dummy unit
+    unit = FactoryBot.create(:unit)
+
+    # Get a new project from unit
+    new_project = unit.active_projects.first
+    
+    # A student enroll to new_project
+    student = new_project.student
+
+    # Perform Get   new_project.main_convenor_user
+    get with_auth_token "/api/projects/#{new_project.id}", new_project.student
+    
+    # Check if the call success
     assert_equal 200, last_response.status
+
+    # Check the returned details match as new_project
+    assert_equal new_project.id, last_response_body['project_id']
+
+    # check if the details posted match as new_project    
+    response_keys = %w(unit_id campus_id enrolled) 
+    project = Project.find(last_response_body['project_id'])
+    assert_json_matches_model(last_response_body, project, response_keys)
```

Comment:
This line already checks all attributes in line 38-40, please do not duplicate the checks, delete lines 38-40 instead. What should be further checked is only `project_id`